File: JS/SEARCHING/2_BinarySearching.js
// Only work with sorted array
function binarySearch(arr, x, lowIndex, highIndex) {
  if (x > arr[highIndex - 1]) {
    console.log("number you search is too big");
    return -1;
  }

  if (highIndex > lowIndex) {
    mid = Math.floor((lowIndex + highIndex) / 2);

    if (arr[mid] > x) {
      return binarySearch(arr, x, 0, mid);
    }
    if (arr[mid] < x) {
      return binarySearch(arr, x, mid, highIndex);
    }
    if (arr[mid] == x) {
      console.log(mid);
      return mid;
    }
  } else {
    console.log("ousides");
    return -1;
  }
}

binarySearch([1, 2, 3, 4, 5, 7, 8, 9, 10], 9, 0, 8);

File: python/SEARCHING/2_binary_searching.py
import math

def binary_searching(arr, x, low_index, high_index):
  if (low_index < high_index):
    mid = math.ceil( (low_index + high_index)/2 )
    if (x < arr[mid]):
      return binary_searching(arr, x, low_index, mid)
    if (x > arr[mid]):
      return binary_searching(arr, x, mid, high_index)
    if(arr[mid] == x):
      return mid
  return -1

print(binary_searching([1,2,3,4,5,6], 5, 0, 5))